Runtime/Scripts/BaseMeshSync.Textures.cs

```diff
@@ -0,0 +1,258 @@
+
+            // LoadAssetAtPath can throw an exception if it's called during a domain backup but we have no way to detect that:
+            try {
+                savedTexture = AssetDatabase.LoadAssetAtPath<Texture2D>(savePath);
+                if (savedTexture == null) {
+                    EditorApplication.delayCall -= UpdatePendingMaterials;
+                    EditorApplication.delayCall += UpdatePendingMaterials;
+
+                    return false;
+                }
+            }
+            catch (UnityException) {
+                return false;
+            }
+
+            mat.SetTexture(textureNameID, savedTexture);
+
+            if (textureNameID == MeshSyncConstants._MetallicGlossMap ||
+                textureNameID == MeshSyncConstants._MaskMap) {
+                mat.EnableKeyword(MeshSyncConstants._METALLICGLOSSMAP);
+                mat.EnableKeyword(MeshSyncConstants._METALLICSPECGLOSSMAP);
+
+                TextureImporter importer = (TextureImporter)AssetImporter.GetAtPath(savePath);
+                if (importer != null) {
+                    importer.sRGBTexture         = false;
+                    importer.alphaIsTransparency = false;
+                    importer.ignorePngGamma      = true;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Tries to apply any pending material updates.
+        /// </summary>
+        private void UpdatePendingMaterials() {
+            for (int i = pendingMaterialUpdates.Count - 1; i >= 0; i--) {
+                Tuple<Material, int> pendingMaterial = pendingMaterialUpdates[i];
+                if (SetSerializedTextureForMaterial(pendingMaterial.Item1, pendingMaterial.Item2)) {
+                    pendingMaterialUpdates.RemoveAt(i);
+                }
+            }
+        }
+#endif
+
+        /// <summary>
+        /// Returns the default shader for the active render pipeline.
+        /// </summary>
+        private static Shader GetStandardShader() {
+#if AT_USE_HDRP
+            return Shader.Find("HDRP/Lit");
+#elif AT_USE_URP
+            return Shader.Find("Universal Render Pipeline/Lit");
+#else
+            return Shader.Find("Standard");
+#endif
+        }
+
+        /// <summary>
+        /// Sets the required shader for the given material.
+        /// </summary>
+        /// <param name="mat">Material to set the shader on</param>
+        /// <param name="shaderName">The name of the shader passed by the DCC tool</param>
+        private static void UpdateShader(ref Material mat, string shaderName) {
+            Shader shader = null;
+            if (!string.IsNullOrEmpty(shaderName)) {
+                shader = Shader.Find(shaderName);
+            }
+
+            bool shaderExists = shader != null;
+
+            if (shader == null) {
+                shader = GetStandardShader();
+            }
+
+            Assert.IsNotNull(shader);
+
+            if (mat == null) {
```

Comment:
Related to my comment in `CreateDefaultMaterial()` function.
I guess what I don't like about this function, is that it allows a new material creation instead of just updating it, which feels "hacky" to me.

Reply:
Creating a new material or updating an existing one use the same code path in `UpdateShader`. The shader can change for a material after it was created or it can be created with a specific shader and change later, this is all handled in a single method.

Reply:
Right, I think this function itself is fine, but I'd prefer an explicit Material creation in `CreateDefaultMaterial()` .

Perhaps that will be possible if we split this function into two ?
1. a function that decides the shader name
2. a function that update the material properties based on the shader name

Reply:
Changed it now.